### PR TITLE
Avoid frequent allocation of little vectors of pointers in stretch

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: *(direct link to the issue)*

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by saintmatthieu** [comment]
Looks fine to me, but I'd refer to @crsib to verify that the changes are worthwhile.

**Comment by crsib** [comment]
If we want to upstream changes - probably we probably shouldn't merge this PR. `calloc` is frowned upon, probably StaffPad guys won't like it either.

**Review by crsib — APPROVED** [review]
I see no performance improvement and I do have doubts that SP guys would want this change, but I have nothing to say against this PR

**Comment by Paul-Licameli** [comment]
Rebased and redrafted without `alloca`

Please review again.

File: libraries/lib-time-and-pitch/StaffPadTimeAndPitch.cpp [hunk 1]
```diff
@@ -82,9 +79,9 @@ void StaffPadTimeAndPitch::GetSamples(float* const* output, size_t outputLen)
          const auto numSamplesToGet =
             std::min({ maxBlockSize, numOutputSamplesAvailable,
                        static_cast<int>(outputLen - numOutputSamples) });
-         const auto buffer =
-            GetOffsetBuffer(output, mNumChannels, numOutputSamples);
-         mTimeAndPitch->retrieveAudio(buffer.data(), numSamplesToGet);
+         float *buffer[2]{};
```
Comment (crsib) [inline libraries/lib-time-and-pitch/StaffPadTimeAndPitch.cpp:82]: Some sanity checks that mNumChannels <= 2 won't hurt, `assert` will do

Comment (Paul-Licameli) [reply libraries/lib-time-and-pitch/StaffPadTimeAndPitch.cpp:84]: Done

**Review by crsib — APPROVED** [review]

**Pull request merged**